programs/KJVDataDump/main.cpp

```diff
@@ -371,14 +396,44 @@ int main(int argc, char *argv[])
 						if (bOutputTemplates) {
 							std::cout << pVerse->m_strTemplate.toUtf8().data() << std::endl;
 						}
-						if (bOutputVerseText) {
-							if (bOutputTemplates) std::cout << strSpacer.toUtf8().data();
-							if (bOutputTransChangeAdded) {
-								std::cout << lstVerseWords.join(QChar(' ')).toUtf8().data();
-							} else {
-								std::cout << strParsedVerse.toUtf8().data();
+						if (bOutputVerseText && nLetterCount > 0) {
+							QString verse = strParsedVerse.toUtf8().data();
+							QString clean = QString(verse).toLower().simplified().replace(QRegularExpression("[\\[\\]()\.,:;!?]"), "");
+							QString cleanNoSpaces = QString(clean).remove(' ');
+							int basicVowelCount = clean.count(QRegularExpression("[aeiou]"));
+							int yCount = clean.count(QRegularExpression("y$|y "));
+							int consonantCount = cleanNoSpaces.length() - (basicVowelCount + yCount);
+
+							bool qPrint = true;
+
+							if (cleanNoSpaces.length() == nLetterCount) {
+
+								QStringList list = clean.split(QRegularExpression("\\s"));
+
+								foreach(QString item, list) {
+									int isYbetween2Consonants = item.count(QRegularExpression("[bcdfghjklmnpqrstvwxz]y[bcdfghjklmnpqrstvwxz]"));
```

Comment:
If such a search type were to be added more generically to KJPBS, a better place to define the letter typing, rather than hardcoding here in this tool what is or isn't a vowel or consonant, would be to add some functionality to the [`ParseSymbols.cpp`](https://github.com/dewhisna/KingJamesPureBibleSearch/blob/master/programs/KJVCanOpener/ParseSymbols.cpp) code.

That way, any of the KJPBS tool suite could make use of the vowel/consonant functionality and it could become more of a generic search criteria.

I could see adding a function to the `StringParse` namespace defined by that code file like `bool isVowel(const QChar &aChar);` or something that could decide if a letter is a vowel or a consonant -- maybe even an enum if there are more than just two types in some foreign languages.

It could/should probably tie in to the translation/language logic so that it also works correctly for foreign languages, as what other languages call a "vowel" are different.  Take Hebrew, for example, it doesn't technically have any vowels.  But some letters sort of behave as vowels.  And modern Hebrew texts make use of vowel markings, not to mention cantillation markings, which could even further broaden the scope of what something like this could search for.

And even English has the 'y' which sometimes acts like a vowel, which I assume is what your "Y between two consonants" logic is about.  To become a staple part of KJPBS, it would need to get generalized and abstracted for foreign language texts.

Reply:
Gotcha.  My initial interest in this PR was just to see if I could get it to do my bidding.  Your review comments are **VERY** helpful as I respect your knowledge and information.  I understand that there is a **proper way** to do things which would take into consideration all different languages supported and there is the **expedient way** to do things which can take much less time and require much less research into languages.  My primary interest is on the numeric phenomenon found in the English KJB text and not so much in applying generalized multi-language parsing logic to the existing search tools.  From your comments, it seems to me that the overall design goals for this project are much broader than simply the analysis of the KJB text.

I will take your suggestions and re-work this code as I find it useful for my purposes but I think for now it would be better to create a separate tool rather than modifying the existing `kjvDump` tool.